agent/csi/mod_csi_docker.go

```diff
@@ -262,10 +262,7 @@ func (d *dockerBridge) cacheListResult(vols []*csi.VolumeInfo) {
 	d.byNameRWL.Lock()
 	defer d.byNameRWL.Unlock()
 
-	// If this is not CSI-NFS then replace the existing volume info objects.
-	if !d.isCSINFS() {
-		d.byName = map[string]csi.VolumeInfo{}
-	}
+	d.byName = map[string]csi.VolumeInfo{}
```

Comment:
I removed the guard around this because this function is now only called when it isn't CSI-NFS.

Reply:
I need to review the rest of the code, but why? Don't we want to cache volumes supplied as the result of a list volumes call?